model/godleyIcon.h

```diff
@@ -115,7 +116,23 @@ namespace minsky
     
     GodleyIcon* clone() const override {
       auto r=new GodleyIcon(*this);
-      r->update();
+      // create a dummy group for the cloned Godley icon to live in while the update takes place. for tickets 1180-1183. 
+      if (auto g=r->group.lock())
+      {
+	   GroupPtr dummy(new Group);
+       for (auto& i: g->items)
+       {		
+          dummy->addItem(i);			  
+          assert(!i->ioVar());
+       }
+	   for (auto& it: dummy->items) 
+	     if (auto godley=dynamic_cast<GodleyIcon*>(it.get())) {
+	        godley->update();
+	        r=godley;
```

Comment:
This leaks the original object allocated on line 118. Plus, the reference obtained on line 129 is most likely invalidated by the clear() call on line 133.

Reply:
Yikes, I thought this would be the only dubious line. I guess a shared pointer would be the way to go? Then I wouldn't have to create a new dummy group, but just share g with it, without deleting g?